types/txn.go

```diff
@@ -315,6 +316,29 @@ func (ctx *TxParseContext) ParseTransaction(payload []byte, pos int, slot *TxSlo
 		}
 		p = dataPos + dataLen
 	}
+	if slot.Type == BlobTxType {
+		p, err = rlp.U256(payload, p, &slot.DataFeeCap)
```

Comment:
You could check if slot.Creation == true here and return error if so.

Reply:
Arguably it's a semantic restriction rather than a parsing error. I added this check to `validateTx` in TxPool.